src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -97,43 +98,55 @@ public async Task<IDelayedException> TryExecuteAsync(IFunctionInstance functionI
                 }
 
                 using (_resultsLogger?.BeginFunctionScope(functionInstanceEx, HostOutputMessage.HostInstanceId))
-                using (parameterHelper)
                 {
-                    try
-                    {
-                        parameterHelper.Initialize();
-                        instanceLogEntry = CreateInstanceLogEntry(functionStartedMessage);
-                        await _functionEventCollector.AddAsync(instanceLogEntry);
-
-                        functionStartedMessageId = await ExecuteWithLoggingAsync(functionInstanceEx, functionStartedMessage, instanceLogEntry, parameterHelper, logger, cancellationToken);
-                    }
-                    catch (Exception exception)
+                    Activity activity = null;
+                    // If no current activity exists, create one for the entire function run.
+                    // HTTP, Service Bus, Event Hub, and other instrumented triggers will have their own activities.
+                    // BeginFunctionScope creates a function activity when AppInsights SDK is enabled.
+                    // In OTel mode, Activity.Current will be null unless the trigger is instrumented.
+                    if (Activity.Current is null)
                     {
-                        functionStartedMessage.Failure = FunctionFailure.FromException(exception);
-                        exceptionInfo = ExceptionDispatchInfo.Capture(exception);
-                        exceptionInfo = await InvokeExceptionFiltersAsync(parameterHelper.JobInstance, exceptionInfo, functionInstanceEx, parameterHelper.FilterContextProperties, logger, cancellationToken);
+                       activity = activitySource.StartActivity(functionInstanceEx.FunctionDescriptor.LogName, ActivityKind.Server);
```

Comment:
Could this go before line 100? That then avoids the extra indentation level for the `using (parameterHelper)` block.

You could also move the activity creation to a helper (which returns `null` if one already exists), so you can also have it in a `using` block.

Reply:
Moving it before line 100 i.e., `BeginFunctionScope` will break the existing AppInsights SDK integration.
Moved it to a helper and using `using` block now.